### PR TITLE
Fix App-IDs across examples and tests

### PR DESCRIPTION
**Pull request by shubham1172**
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

# Description

Fix breaking changes due to https://github.com/dapr/dapr/pull/5556

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: sdk-tests/src/test/java/io/dapr/it/DaprRun.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ private DaprRun(String testName,
                   DaprApiProtocol protocol,
                   DaprApiProtocol appProtocol) {
     // The app name needs to be deterministic since we depend on it to kill previous runs.
-    this.appName = serviceClass == null ? testName : String.format("%s_%s", testName, serviceClass.getSimpleName());
+    this.appName = serviceClass == null ? testName : String.format("%s-%s", testName, serviceClass.getSimpleName());
```
Comment (pravinpushkar) [inline sdk-tests/src/test/java/io/dapr/it/DaprRun.java:61]: @shubham1172 Is this change related to the change done in runtime ? That change only talk about `.` or empty string if I remember correctly.

Comment (pravinpushkar) [reply sdk-tests/src/test/java/io/dapr/it/DaprRun.java:61]: Ya, I think it does prohibit "_" also.

**Review by pravinpushkar — DISMISSED** [review]

File: sdk-tests/src/test/java/io/dapr/it/DaprRun.java (hunk 1 above)
Comment (shubham1172) [reply sdk-tests/src/test/java/io/dapr/it/DaprRun.java:61]: It does talk about `.` but it does prohibit `_` and upper case letters too.

**Review by mukundansundar — APPROVED** [review]
LGTM.
Irrespective of https://github.com/dapr/dapr/pull/5556, I think this change is correct. 
cc @artursouza thoughts?

**Review by pravinpushkar — APPROVED** [review]

**Comment by shubham1172** [comment]
@mukundansundar can we merge this?
